backend/onyx/db/users.py

```diff
@@ -110,13 +113,84 @@ def list_users(
     return db_session.scalars(stmt).unique().all()
 
 
+def _get_accepted_user_where_clause(
+    email_filter_string: str = "",
+    status_filter: UserStatus | None = None,
+    roles_filter: list[UserRole] = [],
+    include_external: bool = False,
+) -> list:
+    # Init where clause and remove any users with API email domains
+    where_clause = [not_(User.email.endswith(DANSWER_API_KEY_DUMMY_EMAIL_DOMAIN))]  # type: ignore
+
+    # Exclude external permissioned users
+    if not include_external:
+        where_clause.append(User.role != UserRole.EXT_PERM_USER)
+
+    if email_filter_string:
+        where_clause.append(User.email.ilike(f"%{email_filter_string}%"))  # type: ignore
+
```

Comment:
Used "type: ignore" here referencing what the preexisting function fetching users did